doc/cpd/61.rst

```diff
@@ -0,0 +1,193 @@
+ssh_priv_key The ssh private key to use for accessing the ``Host``.
+remote_user  The username to use with the ``ssh_priv_key`` when accessing the ``Host``.
+============ ===========
+
+To accommodate the possibility of external ``Host`` instances the following field would be added.
+
+================= ===========
+Name              Description
+================= ===========
+external_provider Name of the external system of record that can authoritatively provide ``Host`` information.
+================= ===========
+
+The use of ``external_provider`` will determine if the ``Host`` instance should be populated from
+the general store defined by commissaire, or a specific store.
+
+* When ``external_provider`` is not defined, the ``Host`` record is considered native to Commissaire.
+* If the ``external_provider`` is defined it should be the name of the StoreHandler which can provide
+the ``Host`` information. For instance, if ``commissaire.storage.cloudforms`` should be used, then the
+``external_provider`` would be ``cloudforms``.
+
+
+Example Host Model Owned by Commissaire
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: javascript
+
+   {
+       "space": 51475068,
+       "status": "active",
+       "address": "192.168.155.150",
+       "os": "fedora",
+       "memory": 2048,
+       "cpus": 4,
+       "last_check": "2016-11-28T22:10:11.851787",
+       "external_provider": ""
+   }
+
+
+Example Host Model Owned by An External Provider
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note::
+
+    All data but ``status``, ``last_check``, and ``external_provider`` would come from the external_provider.
+
+.. code-block:: javascript
+
+  {
+      "space": 51475068,
+      "status": "active",
+      "address": "192.168.155.150",
+      "os": "fedora",
+      "memory": 2048,
+      "cpus": 4,
+      "last_check": "2016-11-28T22:10:11.851787",
+      "external_provider": "cloudforms"
+  }
+
+
+Accessing Hosts
+~~~~~~~~~~~~~~~
+``StorageService`` will still be the authoritative service for retrieving ``Host`` data.  For
+``StorageService`` to be able to make these external calls a ``StoreHandler`` would need to be
+available and configured for any ``external_provider`` in use. As an example of a ``StoreHandler``
+see the `etcd StoreHandler <https://github.com/projectatomic/commissaire/blob/master/src/commissaire/storage/etcd.py>`_.
+
+
+Changes to StorageService
+~~~~~~~~~~~~~~~~~~~~~~~~~
+``StorageService`` currently only allows one ``StoreHandler`` to be configured per model
+(See `this code chunk <https://github.com/projectatomic/commissaire-service/blob/339b4f6aabcbdee78ae9d24160e9974166492503/src/commissaire_service/storage/storehandlermanager.py#L73-L86>`_).
+This restriction would need to be changed so that multiple ``StoreHandlers`` can be configured with a model.
+A new parameter defining a ``StoreHandler`` as the default for the models it has configured woul need to
+be added.
```

Comment:
Hmm... could we just say the first `StoreHandler` defined for a particular model type is its default?  I suspect that's going to be most common, and *if* it proves necessary we could add the `default` parameter later as an override.

I'm just thinking about what happens we someone (namely, me) inevitably forgets to tag one of the `StoreHandler` definitions as default.  I think we need a reasonable implicit policy.